src/Questar.OneRoster/Filtering/FilterValueBuilder.cs

```diff
@@ -0,0 +1,100 @@
+namespace Questar.OneRoster.Filtering
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Linq.Expressions;
+    using System.Reflection;
+    using Common;
+    using Reflection;
+
+    public sealed class FilterValueBuilder : ExpressionVisitor
+    {
+        public FilterValueBuilder(PropertyInfo property)
+            : this(property, null)
+        {
+        }
+
+        public FilterValueBuilder(PropertyInfo property, Expression expression)
+        {
+            Property = property;
+            Expression = expression;
+        }
+
+        public Expression Expression { get; }
+
+        public PropertyInfo Property { get; }
+
+        public FilterValue Value { get; private set; }
+
+        public override Expression Visit(Expression node)
+        {
+            if (IsTerminal(node)) return node;
+
+            switch (node.NodeType)
+            {
+                case ExpressionType.Constant:
+                    return base.Visit(node);
+                default:
+                    Value = GetValue(Evaluate(node));
+                    return node;
+            }
+        }
+
+        private static object Evaluate(Expression node)
+        {
+            var convert = Expression.Convert(node, typeof(object));
+            var lambda = Expression.Lambda<Func<object>>(convert);
+            var getter = lambda.Compile();
+            var value = getter();
+            return value;
+        }
+
+        private bool IsTerminal(Expression expression)
+            => Expression?.Equals(expression) == true;
+
+        private string FormatScalar(object value)
+        {
+            var property = Property.PropertyType;
+            var converter = property.GetConverter();
+            if (converter == null)
+                throw new InvalidOperationException($"Couldn't find type converter for type '{property}'");
+
+            // TODO find a better solution for formatting in general...
```

Comment:
I'd rather treat it as another feature. It may be a bit larger in scope than I'd like for a single PR. This is already pretty large. Basically, we need a way to resolve the value format, if applicable. I'm thinking either an attribute, custom type converter implementation, or both.